FreeRTOS/FreeRTOS-Kernel
# Adding SMP coverity example

Adding SMP coverity example

Description
-----------
In this PR:
* Add SMP coverity example. User can reference the ./examples/coverity/README.md to run coverity with SMP FreeRTOS.
* Suppress MISRA C:2012 directive 4.7 by comment. The return value of `taskENTER_CRITICAL_FROM_ISR()` is the interrupt mask. No error information will be returned by this function and no need to perform a check.

Test Steps
-----------
Run coverity 2023.6.1 with the following command should shows no violations.
Single core
```
cmake -B build -S examples/coverity
```
SMP
```
cmake -B build -S examples/coverity -DFREERTOS_SMP_EXAMPLE=1
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
